crates/sui/src/client_commands.rs

```diff
@@ -1000,19 +1043,38 @@ impl SuiClientCommands {
                         skip_dependency_verification,
                     )
                     .await?;
-
-                let data = client
+                let tx_kind = client
                     .transaction_builder()
-                    .upgrade(
-                        sender,
+                    .upgrade_tx_kind(
                         package_id,
                         compiled_modules,
                         dependencies.published.into_values().collect(),
                         upgrade_capability,
                         upgrade_policy,
                         package_digest.to_vec(),
-                        gas,
+                    )
+                    .await?;
+                if dry_run {
+                    return execute_dry_run(
+                        context,
+                        sender,
+                        tx_kind,
+                        gas_budget,
+                        gas_price,
+                        gas.map(|x| vec![x]),
+                        None,
+                    )
+                    .await;
+                }
+                let data = client
+                    .transaction_builder()
+                    .tx_data(
+                        sender,
+                        tx_kind,
                         gas_budget,
+                        gas_price,
+                        gas.map(|x| vec![x]).unwrap_or_default(),
```

Comment:
You can do this as follows instead:

```suggestion
                        gas.into_iter().collect(),
```

Reply:
(Same goes for all the  examples below).